design/rook-ceph-status-conditions.md

```diff
@@ -0,0 +1,93 @@
+# Rook Ceph Operator Status Condition
+
+This work is heavily based on: https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md
+
+## Background
+
+Currently, Ceph Cluster stores the state of the system in `Status.State`. But, we want to implement the usage of `Status.Conditions` instead of using `Status.State`. The usage of `Status.Phase` is deprecated over time because it contradicts the system design principle and hampered evolution. So rather than encouraging clients to infer the implicit properties from phases, the usage of `Status.Condition` is preferred. Conditions are more extensible since the addition of new conditions doesn't invalidate decisions based on existing conditions, and are also better suited to reflect conditions that may toggle back and forth and/or that may not be mutually exclusive.
+
+## Conditions
+
+Conditions simply represent the latest available observation of an object's state. They are an extension mechanism intended to be used when the details of observation are not a priori known or would not apply to all instances of a given Kind. Objects can have multiple Conditions, and new types of Conditions can also be added in the future by the third-party controllers. Thus, Conditions are thereby represented using list/slice, where each having the similar structure.
+
+## System States for rook-ceph
+
+The necessary system states for the rook-ceph can be portrayed as follows:
+	   
+	   Ignored 		: If any of the resources gets ignored for multiple reasons
+	   Progressing 		: Marks the start of reconcile of Ceph Cluster
+	   Ready 		: When Reconcile completes successfully
+	   Not Ready 		: Either when cluster is Upgrading or Upgrading is blocked
+	   Connecting		: When the Ceph Cluster is in the state of Connecting 
+	   Available 		: The Ceph Cluster is healthy and is available to use
+	   Failure 		: If any failure occurs in the Ceph Cluster
+	   Cluster Expanding	: If the Cluster is Expanding
+
+## Implementation Details
+
+Implementation as Specified in https://github.com/openshift/custom-resource-status/:
+
+We can add Conditions simply in the Custom Resource struct as:
+	  
+	   type ClusterStatus struct{
+		Conditions     *RookConditions    `json:"conditions,omitempty"`
```

Comment:
There can be multiple conditions, right? Should this be a slice?

Reply:
Yes. It was suppossed to be a slice. I changed it in my document. Will push it in.